packages/next-on-pages/src/buildApplication/processVercelFunctions/dedupeEdgeFunctions.ts

```diff
@@ -168,6 +176,32 @@ async function processFunctionIdentifiers(
 	await Promise.all(functionBuildPromises);
 }
 
+/**
+ * Given a standard ESM file (without imports) it converts it to a function call that returns
+ * an object with the various exports set as its fields
+ *
+ * The function allows us to override global symbols such as `self`, `globalThis` and `global`
+ * (which are used as the function's parameter names)
+ *
+ * @param path the path of the ESM file
+ * @returns the converted file content
+ */
+async function functionifyFileContent(path: string) {
+	let fileContents = await readFile(path, 'utf8');
+	fileContents = `const namedExports = {};${fileContents}`;
+	fileContents = fileContents.replace(
+		/export const (\S+) =/g,
+		'namedExports["$1"] =',
```

Comment:
i'm not sure this will work quite as well as we're hoping.

I can't seem to find any function files that have `export const ... =` in them in either of the projects i've looked in, so im not sure where this would be updating things?

It also looks like this would overwrite the next.js error messages and add them to the named exports object which probably isn't ideal.

<img width="828" alt="image" src="https://github.com/user-attachments/assets/78090cff-56a9-4b2e-9c9e-1fafd1e5a0c7">

Reply:
> I can't seem to find any function files that have export const ... = in them in either of the projects i've looked in, so im not sure where this would be updating things? Unless it's only certain use cases this would apply to?

mh... I'll have a look 👀 

___

I'm not sure I understand the comment about the Next.js error messages... could you clarify what you mean?

Reply:
Are you sure the regex replacement doesn't do anything? I just created a brand new next-on-pages app using C3 and tried to run this branch's build with the regex replacement removed and I immediately get these build errors:
![Screenshot 2024-07-31 at 23 36 05](https://github.com/user-attachments/assets/dd9fd1ef-14e9-4358-96f0-567f85863ebb)

To be fair these are exports next-on-pages itself adds... so maybe you only checked in the Vercel output files?

Reply:
> could you clarify what you mean?

the regex would try and add an entry to the object and replace part of the error message

> To be fair these look like exports that next-on-pages itself adds/creates... so maybe you only checked in the Vercel output files?

i haven't run a build with this, but i'm looking at a build of cloudy from about a month ago, and i can't find `export const` anywhere in the function files (vercel or next-on-pages).

vercel functions:
<img width="833" alt="image" src="https://github.com/user-attachments/assets/079c4522-797e-41c1-905d-69b84d53a2ec">

_worker.js dir:
<img width="845" alt="image" src="https://github.com/user-attachments/assets/204712be-b230-4a99-815f-2e221ba42e47">

Reply:
the suspense cache isn't one - that's just a variable we use internally. but there's plenty of references to global, self, etc. in these function files but none of them would go through the proxy

Reply:
> the suspense cache isn't one

yeah no, my bad, we pass the cache files trhough the functions build pipeline

> but there's plenty of references to global, self, etc. in these function files but none of them would go through the proxy

what do you mean? could you clarify? why would they not go through the proxy?

Reply:
actually nevermind. i think im just misunderstanding how the named exports and the function wrapping the file is used

Reply:
Note: We do get `export const` in at this step, especially since we've added the lines outselves earlier:
https://github.com/cloudflare/next-on-pages/blob/0a07f340b868bb887e465b86557efdae793c348f/packages/next-on-pages/src/buildApplication/processVercelFunctions/dedupeEdgeFunctions.ts#L386

The fact that we don't get them in the final output, before this PR, I believe is because those get converted by esbuild to exports-as: 
![Screenshot 2024-08-01 at 13 16 11](https://github.com/user-attachments/assets/fb5e3a51-ad47-475d-87ce-71375999d4e7)

I am really wondering if we could not add the `export const`s ourselves.... but by debugging this I noticed that there are indeed a few `export const`s that we're not adding... so I might need to keep things as they are unfortunately 😕

Reply:
> actually nevermind. i think im just misunderstanding how the named exports and the function wrapping the file is used

no come on, if you have any concern I am happy to hear them 🙂🙏 

shall we move this async and discuss things?

Reply:
@james-elicx regex replacement removed! 🚀 https://github.com/cloudflare/next-on-pages/pull/836/commits/b13338aebe83572a3e34a9cf9221e765afaf0547

The less of these we have the better! thanks a lot for pointing this and making me improve the solution! ❤️

Reply:
🚀 

![Screenshot 2024-08-01 at 17 29 15](https://github.com/user-attachments/assets/20b42048-cce8-4c63-8970-67193175b7bf)

Reply:
Resolved as discussed async 😄